modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -154,9 +154,11 @@ class PandasDataframe(ClassLogger):
     ----------
     partitions : np.ndarray
         A 2D NumPy array of partitions.
-    index : sequence, optional
+    index : sequence or callable, optional
```

Comment:
Should we add similar logic for columns?

Reply:
"A new callable type for `_columns_cache` can be easily added, but I don't want to do this until there is code that will use this feature."